Cargo.toml

```diff
@@ -48,17 +48,23 @@ usbd-serial = "0.1.0"
 [features]
 default = ["unproven"]
 device-selected = []
+needs-subvariant = []
 rt = ["stm32f3/rt"]
-stm32f301 = ["stm32f3/stm32f301", "device-selected"]
+# Any Changes here should be mirrored in README.md and src/lib.rs
+stm32f301 = ["stm32f3/stm32f301", "needs-subvariant"]
+stm32f301xb = ["stm32f301", "device-selected"]
+stm32f301xc = ["stm32f301", "device-selected"]
+stm32f301xd = ["stm32f301", "device-selected"]
+stm32f301xe = ["stm32f301", "device-selected"]
 stm32f318 = ["stm32f3/stm32f3x8", "device-selected"]
 stm32f302 = ["stm32f3/stm32f302", "device-selected"]
-stm32f303 = ["stm32f3/stm32f303", "device-selected"]
-stm32f303xb = ["stm32f303", "stm32-usbd/ram_access_1x16"]
-stm32f303xc = ["stm32f303", "stm32-usbd/ram_access_1x16"]
-stm32f303xd = ["stm32f303", "stm32-usbd/ram_access_2x16"]
-stm32f303xe = ["stm32f303", "stm32-usbd/ram_access_2x16"]
-stm32f303x6 = ["stm32f303"]
-stm32f303x8 = ["stm32f303"]
+stm32f303 = ["stm32f3/stm32f303", "needs-subvariant"]
+stm32f303xb = ["stm32f303", "stm32-usbd/ram_access_1x16", "device-selected"]
```

Comment:
It's just that we are using the `#[cfg(any(feature = "stm32f303",…))]` in a lot of places. I thought it would be a little too verbose to use the `stm32f3/stm32f303` everywhere but I also see your point.